devmgmtV2/controllers/cloud.controller.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+let request = require('request-promise');
+let moment = require('moment');
+let {
+	config
+} = require('../config'); // get config.js
+
+let getState = () => {
+	let {
+		userToken,
+		authToken,
+		baseUrl,
+		searchUrlSuffix,
+		filter
+	} = config.cloudAPI;
+
+	let { keysToUse } = filter;
+
+	// console.log({config});
+
+	return {
+		userToken: (token) => {
+			userToken = token || userToken;
+			return userToken;
+		},
+
+		authToken: (token) => {
+			authToken = token || authToken;
+			return authToken;
+		},
+
+		searchUrl: (suffix) => {
+			searchUrlSuffix = suffix || searchUrlSuffix;
+			return `${baseUrl}${searchUrlSuffix}`;
+		},
+
+		keysToUse: (keys) => {
+			keysToUse = keys || keysToUse;
+			return keysToUse;
+		}
+	};
+};
+
+let getTimestamp = (pattern) => moment().format(pattern);
+
+let getHeaders = () => {
+	const pattern = 'YYYY-MM-DD HH:mm:ss:SSSZZ';
+	const timestamp = getTimestamp(pattern);
+
+	const state = getState();
+
+	return {
+		'Accept': 'application/json',
+		'Content-Type': 'application/json',
+		'X-Consumer-ID': '029c72b5-4691-4bf2-a6de-72b18df0b748',
+		'ts': timestamp,
+		'X-msgid': '8e27cbf5-e299-43b0-bca7-8347f7e5abcf',
+		'X-Device-ID': 'X-Device-ID',
+		'X-Authenticated-User-Token': state.userToken(),
+		'X-Source': 'app',
+		'Authorization': `Bearer ${state.authToken()}`
+	};
+};
+
+let getSearchBody = (queryString) => {
+	const pattern = 'YYYY/MM/DD HH:mm:ss';
+	const timestamp = getTimestamp(pattern);
+
+	return {
+		'id': 'unique API ID',
+		'ts': timestamp,
+		'params': {
+		},
+		'request': {
+			'query': queryString,
+			'filters': {
+				'contentType': ['Collection', 'Story', 'Worksheet', 'TextBook', 'Course', 'LessonPlan', 'Resource']
+			}
+		}
+	};
+};
+
+let getRequestOptions = (method, uri, body, headers, json = true) => ({
+	method,
+	uri,
+	body,
+	headers,
+	json // Automatically stringifies the body to JSON
+});
+
+let searchSunbirdCloud = (queryString = '') => {
+	const state = getState();
+	const body = getSearchBody(queryString);
+	const headers = getHeaders();
+	const uri = state.searchUrl();
+	const options = getRequestOptions('POST', uri, body, headers);
+
+	return request(options);
+};
+
+let filterObjectKeys = (obj = {}, keysToUse = Object.keys(obj)) => {
+	let filteredObj = {};
+	keysToUse.forEach(key => filteredObj[key] = obj[key]);
+	return filteredObj;
+}
+
+let filterKeysInObjectList = (results, keysToUse) => {
+	return results.map(obj => filterObjectKeys(obj, keysToUse));
```

Comment:
can use .filter chain here itself... no ?

Reply:
I thought I'll move that out. Felt more to clean to me and in case I need to do it for a single object somewhere I have it handy. Is it a bad idea?

Reply:
No..It's okay..I thought you forgot to chain it :laughing: .